src/main/kotlin/Utilities/JsonGenerator.kt

```diff
@@ -1,20 +1,16 @@
 package Utilities
 
 import com.google.gson.Gson
-import org.json.JSONObject
+import kotlin.reflect.KClass
 
 class JsonGenerator {
-    private val jsonObj = JSONObject()
     private val gson = Gson()
 
-    fun generateJsonString(mapOfKeysToValues: Map<String, Any>): String {
-        for (entries in mapOfKeysToValues) {
-            jsonObj.put(entries.key, entries.value)
-        }
-        return jsonObj.toString()
-    }
-
     fun resourceToJson(resource: Any): String {
         return gson.toJson(resource)
     }
+
+    fun <T: Any> resourceFromJson(resource: String, obj: KClass<T>): T {
```

Comment:
what do you mean by a resource? Is it a JSON string?

Reply:
same question for `resourceToJson`. Does it convert a JSON to a JSON string?

Reply:
By resource I mean a domain concept that contains some data, like `Task`. `resourceFromJson` means converting a string formatted `Json` into a resource, and `resourceToJson` means converting a resource into a string formatted `Json`. Do you think renaming it from resource to object would help with readability?